java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -655,6 +662,19 @@ else if (action.getActionType().equals(ActionFactory.TYPE_HARDWARE_REFRESH_LIST)
                     .ifPresent(minionServer -> handleHardwareProfileUpdate(minionServer, Json.GSON.fromJson(jsonResult,
                     HwProfileUpdateSlsResult.class), serverAction));
         }
+        else if (action.getActionType().equals(ActionFactory.TYPE_VIRT_PROFILE_REFRESH)) {
+            if (serverAction.getStatus().equals(ActionFactory.STATUS_FAILED)) {
+                var originalErrorMsg = getErrorFromStateApplyResult(jsonResult,
+                        "Error while updating the virtual instance profile.\nGot no result from the system.");
+                serverAction.setResultMsg(originalErrorMsg);
+            }
+            else {
+                serverAction.setResultMsg("Success");
+            }
+            serverAction.getServer().asMinionServer()
```

Comment:
IF we don't find the minion in the database should we log something with this data? It should be a edge case, but can happen.